src/main/java/com/aws/greengrass/authorization/AuthorizationHandler.java

```diff
@@ -419,12 +419,17 @@ private void addPermission(String destination,
                             Permission.builder().principal(principal).operation(operation).resource(null).build());
```

Comment:
So in this case it still propagates the exception? why are we handling differently?

Reply:
I believe that part will never have exception but if in case it does, that exception should propagate up and log `Error while loading policy` instead of `Error while adding resource` because we are not really going through each resource.
So I though it's ok to keep it that way.